zebrad/tests/acceptance.rs

```diff
@@ -240,6 +250,56 @@ fn serialized_tests() -> Result<()> {
     start_no_args()?;
     start_args()?;
     seed_no_args()?;
+    valid_generated_config()?;
+
+    Ok(())
+}
+
+fn valid_generated_config() -> Result<()> {
+    zebra_test::init();
+
+    // Generate configuration in binary parent path
+    let binary_path = Path::new(env!("CARGO_BIN_EXE_zebrad"));
+    let path = format!(
+        "{}/{}",
+        binary_path.parent().unwrap().to_str().unwrap(),
+        "zebrad.toml"
+    );
+    let (child, _guard) = get_child(&["generate", "-o", &path])?;
+    let output = child.wait_with_output()?;
+    output.assert_success()?;
+
+    // Check if the file was created
+    assert_eq!(Path::new(&path).exists(), true);
+
+    // Check if it was created in the same path as binary
+    assert_eq!(Path::new(&path).parent(), binary_path.parent());
```

Comment:
See my suggestions above about `TempDir` and `PathBuf`.

Also, rather than duplicating this code, you might be able to put it in a separate function.